### PR TITLE
APK Backdooring script

### PR DESCRIPTION
**Pull request by Jack64**
POC of an arbitrary APK meterpreter injector.
Most of it was done by @timwr and I added some changes on finding / selecting the hook point .
Keep in mind this requires the latest git version of apktool (v2.0.1 or higher).

### CODE REVIEW

**Comment by void-in** [comment]
LHOST and LPORT is having hard coded values. Also the script assumes it will always be run from $MSF_BASE since it is using ./msfvenom.

**Comment by kernelsmith** [comment]
I admit I haven't looked at the code here, but was wondering if this should folded into msfvenom directly, possibly as a case to the -x? template arg. If the payload is Java and the template is an apk then do this kinda thing?

**Comment by hmoore-r7** [comment]
Moving it to ``msfvenom`` would be the long-term solution, but due to dependencies on external tools, we would need to handle this better. I figure we clean the script up to the point that we can include it as a standard alone utility and then eventually merge it into msfvenom, adding checks for on-system apktool and java versions.

**Comment by Jack64** [comment]
Hmm, I have an update for this script, but I deleted the original repository it was on accidentally (it was a fork of @timwr 's repo, not rapid7's) and now I can't edit this PR.
Should I close this and make a new fork/PR with the update?

**Comment by kernelsmith** [comment]
Did u delete it locally and remotely (on GitHub)? If you did not delete it remotely you can just pull it from there back to your local machine and as usual edit it, commit, and push back up.

If you deleted it remotely too, you could probably clone the PR branch from rapid7 (even PRs are treated as just another branch), but it would probably require enough gymnastics that opening a new PR would be easier.

**Comment by Jack64** [comment]
hey @kernelsmith , I did remove it remotely as well. I could try to clone the PR but I think the only point of that would be to save the discussion above. In any case I do have an update to push so I await instructions as to the best way to go about it.

File: tools/apk_backdoor.rb [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+                return activityname
+            end
+        end
+    end
+end
+
+# If XML parsing of the manifest fails, recursively search
+# the smali code for the onCreate() hook and let the user
+# pick the injection point
+def scrapeFilesForLauncherActivity()
```
Comment (hmoore-r7) [inline tools/apk_backdoor.rb]: Would you mind changing this to normal ruby casing? 
```scrape_files_for_launcher_activity()```

File: tools/apk_backdoor.rb [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+# arbitrary APKs. Latest Git version of Apktool is recommended.
+# Authored by timwr, Jack64
+#
+
+
+require 'nokogiri'
+require 'fileutils'
+
+# Find the activity that is opened when you click the app icon
+def findlauncheractivity(amanifest)
```
Comment (hmoore-r7) [inline tools/apk_backdoor.rb]: This could use a rename to ``find_launch_activity()``

File: tools/apk_backdoor.rb [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+end
+
+apktool = `which apktool`
+unless(apktool && apktool.length > 0)
+	puts "No apktool"
+	exit(1)
+end
+
+apk_v=`apktool`
+unless(apk_v.split()[1].include?("v2."))
```
Comment (hmoore-r7) [inline tools/apk_backdoor.rb]: This should probably tell the user that apktool needs to be in their PATH as well.

**Comment by Jack64** [comment]
Closing this because I deleted the original fork, you can find it @ https://github.com/rapid7/metasploit-framework/pull/5932

**Pull request closed without merging**